### PR TITLE
shows how to wire admission control in a sample api server.

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**:
This PR shows how to wire admission control in a custom api server

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by sttts** [comment]
@k8s-bot ok to test

**Comment by deads2k** [comment]
@p0lyn0mial Can you provide a summary of your findings when you tried this?  I think you uncovered a bug.

**Comment by p0lyn0mial** [comment]
when I enabled `NamespaceLifecycle` plugin I got `Failed to list *v1.Namespace: the server could not find the requested resource (get namespaces)`. I thought that my server simply did not have the resource. 

I talked to @deads2k and he said this might be actually a bug. For namespaces it should talk to the kube-apiserver

**Comment by p0lyn0mial** [comment]
I am looking into this.

**Comment by sttts** [comment]
The admission initializer always uses the loopback client (which is wrong here of course). Good finding!

**Comment by p0lyn0mial** [comment]
@sttts  it is safe to bring back tags. Fix delivered https://github.com/kubernetes/kubernetes/pull/47443#event-1139659784

File: staging/src/k8s.io/sample-apiserver/pkg/cmd/server/start.go [hunk 1]
```diff
@@ -71,7 +79,7 @@ func NewCommandStartWardleServer(out, errOut io.Writer, stopCh <-chan struct{})
 	}
 
 	flags := cmd.Flags()
-	o.RecommendedOptions.AddFlags(flags)
+	o.addFlags(flags)
```
Comment (deads2k) [inline staging/src/k8s.io/sample-apiserver/pkg/cmd/server/start.go]: optional nit: my personal preference is to keep the flag binding here.  I suspect we'll actually move admission into recommended options this release.  When we move it, I'd like to move the `AddFlags` back here.

**Comment by deads2k** [comment]
this fell out really nicely.  

/lgtm
/approve no-issue

**Comment by deads2k** [comment]
@k8s-bot test this

**Pull request merged**